src/MiniProfiler.EF6/MiniProfilerEF6.cs

```diff
@@ -35,5 +46,29 @@ public static void Initialize()
                 }
             }
         }
+
+        private static DbProviderServices WrapDbProviderServices(DbProviderServices inner, object key)
+        {
+            var cacheKey = new { type = typeof(DbProviderServices), key }; // TODO: consider using ValueTuple or implementing a similar polyfill
```

Comment:
I'd use a ValueTuple here normally, but depending on the bounds I'd instead opt for 4 properly typed `ConcurrentDictionary` instead, which means only `key` is used anyway.